tests/compile_upp.sh

```diff
@@ -2,12 +2,40 @@
 # Wen Meng 01/2020, Set up for cmake build.
 #############################################
 
-set -x
+set -eu
 
-#Clean loaded modules
-module purge
+usage() {
+  echo
+  echo "Usage: $0 [-g] [-w] -h"
+  echo
+  echo "  -p  installation prefix <prefix>    DEFAULT: ../install"
+  echo "  -g  Build with GTG                  DEFAULT: OFF"
+  echo "  -w  Build with WRF-IO               DEFAULT: OFF"
```

Comment:
@aerorahul Could you set WRF-IO ON as default?

Reply:
I prefer to keep it this way, because if one has to turn it OFF, what option should the script have?  Besides, `-g` is to turn `GTG` ON and `-w` to turn `WRFIO` OFF are polar opposites.
It is not that cumbersome to do `./compile_upp.sh -w`  instead of `./compile_upp.sh`.

Reply:
@aerorahul A lot of UPP users from rtma, hrrr/rap need the UPP executable built with WRF_IO.  I would image we get UPP issues of failure on processing WRF_IO data due to user forgetting add option "-w" in building.

Reply:
I am going to pull out of the changes to `tests/compile_upp.sh` and let the CODEOWNERS tackle how they see fit in a later PR.
This PR is about adding GTG compile option to CMake and does not change the current default behaviour.

Reply:
Removed changes to `tests/compile_upp.sh`